### PR TITLE
fix json/toml formatting for root dir

### PR DESCRIPTION
**Pull request by alexeldeib**

### CODE REVIEW

**Comment by alexeldeib** [comment]
Compare changes in RP: https://dev.azure.com/msazure/CloudNativeCompute/_git/aks-rp/pullrequestcreate?sourceRef=ace%2Fformat&targetRef=ace%2Fredisk&sourceRepositoryId=45442779-e72f-4447-9cd5-d3d2e2329e6e&targetRepositoryId=45442779-e72f-4447-9cd5-d3d2e2329e6e

I uploaded the diff here for easier viewing if devops doesn't work: https://diffy.org/diff/cv67g8q3q5av8l9cgbsxlxr

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 1]
```diff
@@ -198,9 +198,9 @@ write_files:
   permissions: "0644"
   owner: root
   content: |
-    {{if HasDataDir}}
+    {{if HasDataDir -}}
```
Comment (xuto2) [inline parts/linux/cloud-init/nodecustomdata.yml:201]: curious why this is different than L168?

Comment (alexeldeib) [reply parts/linux/cloud-init/nodecustomdata.yml:201]: good callout, I thought one originally needed extra trim but they are the same. will fix.

**Pull request merged**